src/garrysmod/gamemodes/zombiesurvival/entities/weapons/weapon_zs_basemelee/shared.lua

```diff
@@ -428,3 +428,7 @@ function SWEP:TranslateActivity( act )
 
 	return self.ActivityTranslate and self.ActivityTranslate[act] or -1
 end
+
+function SWEP:IsOwnerBarricadeGhosting()
+	return self:GetOwner():GetBarricadeGhosting()
+end
```

Comment:
I added this so we wouldn't have to copy the `SWEP:CanPrimaryAttack` and `SWEP:GetViewModelPosition` functions. We only need to override `SWEP:IsOwnerBarricadeGhosting` in the subclass to effect behavior in the baseclass.